app/org/sagebionetworks/bridge/dynamodb/DynamoSubpopulationDao.java

```diff
@@ -131,14 +141,15 @@ public Subpopulation getSubpopulation(StudyIdentifier studyId, String guid) {
     }
     
     @Override
-    public void deleteSubpopulation(StudyIdentifier studyId, String guid) {
-        Subpopulation subpop = getSubpopulation(studyId, guid);
+    public void deleteSubpopulation(StudyIdentifier studyId, SubpopulationGuid subpopGuid) {
+        Subpopulation subpop = getSubpopulation(studyId, subpopGuid);
         if (subpop == null || subpop.isDeleted()) {
             throw new EntityNotFoundException(Subpopulation.class);
         }
         if (subpop.isDefaultGroup()) {
             throw new BadRequestException("Cannot delete the default subpopulation for a study.");
         }
+        studyConsentDao.deleteAllConsents(subpopGuid);
```

Comment:
We should add a unit test to verify that consents are deleted when the subpop is deleted.